### DISCUSSION TITLE
Navigation.navigate() with an auto-completable type

### DISCUSSION THREAD

**Discussion by Dohun-choi** [react-navigation/react-navigation#11901] [Ideas]
When passing a screen name as an argument to Navigation.navigate(), users can navigate to the corresponding screen. With the following implementation, type errors are avoided, and developers can utilize autocomplete for the screen names they've defined.

I referred to @react-navigation/routers/src/type.ts, line 97:

export type ParamListBase = Record<string, object | undefined>;

type Navigation = NativeStackNavigationProp<
    // Modifying ParamListBase
    Record<keyof NativeStackParamList, object | undefined>
>;

When using navigate in components, developers can utilize autocomplete for registered screens, facilitating easier code writing. For example, when using useNavigation<Navigation>("screenName"), screen names can be auto-completed.

If such a feature already exists, please disregard my suggestion. It would be better if you could inform me of that method.

I used a translation tool, so the sentence may sound awkward.

**Comment by satya164** [comment]
You'll get autocompletion and type checking for `useNavigation` if you have setup your types as documented here https://reactnavigation.org/docs/typescript#specifying-default-types-for-usenavigation-link-ref-etc

**Reply by Dohun-choi** [reply]
sorry for bothering you
	
	function onPress(moveTo: keyof ScreenProps) {
		navigation.navigate(moveTo);
	}
	
I wnat use like this, but typescript said it's wrong..
could you help me to this?
it's ok if i use like this
```
interface Props {
// 0 and 1 are key of ScreenProps
name: "0" | "1"
}
<Pressable onPress={() => navigation.navigate(name)};></Pressable>
```

**Reply by satya164** [reply]
It's not possible to say what's wrong from such little info. Everything depends on your project, the screens, and their params. But it's possible to pass a screen name and use it for `navigate`:

<img width="446" alt="image" src="https://github.com/react-navigation/react-navigation/assets/1174278/2b168d22-d4b9-41c3-9712-99b4cda013c1">
